### DISCUSSION TITLE
Guidance on Using YSU PBL Scheme in HAFS

### DISCUSSION THREAD

**Discussion by muradme** [hafs-community/HAFS#304] [Q&A] [answered]
Hi everyone,

I am currently working on setting up HAFS and would like to use the YSU PBL scheme instead of the default scheme. I noticed that the YSU scheme is available under the directory:

_sorc/hafs_forecast.fd/FV3/ccpp/physics/physics/PBL/YSU/_

I have a couple of questions regarding this:

Do I need to recompile the code with a different DCCPP_SUITES configuration to activate the YSU PBL scheme?
Are there specific configuration files or namelist variables that need to be modified to enable the YSU scheme, or is it entirely handled during the compilation process?

If anyone has experience with this or any insights on the necessary steps to use YSU in HAFS, I would greatly appreciate your guidance!

Thank you for your time and support.

Best regards,
Murad

**Comment by jdkublnick** [comment]
Hello, muradme! I have reached out to our HAFS SMEs, and we should have an answer for you shortly; thank you for your patience!

**Comment by jdkublnick** [comment, marked as answer]
Hello, muradme! HAFS developers are not currently supporting the YSU PBL scheme on HAFS. Enabling new physics schemes in HAFS can lead to unexpected and/or poor results. It is recommended that users run experiments only with the supported physics suites and physics schemes unless they have an excellent understanding of how these physics schemes work and a specific research purpose in mind for making such changes.

All that being said I can provide you with some resources that might help guide you:

CCCP Discussions (these folks have done HAFS physics updates in the past @SamuelTrahanNOAA): https://github.com/NCAR/ccpp-physics/discussions/categories/q-a

Past HAFS physics PRs:
- https://github.com/hafs-community/HAFS/pull/292
- https://github.com/hafs-community/HAFS/pull/294

SRW FAQ (will not be exactly the same but may provide some guidance): https://ufs-srweather-app.readthedocs.io/en/latest/UsersGuide/Reference/FAQ.html#advanced-how-can-i-add-a-physics-scheme-e-g-ysu-pbl-to-the-ufs-srw-app
